langchain/text_splitter.py

```diff
@@ -169,10 +191,17 @@ def __init__(self, encoding_name: str = "gpt2", **kwargs: Any):
         # create a GPT-3 encoder instance
         self._tokenizer = tiktoken.get_encoding(encoding_name)
 
-    def split_text(self, text: str) -> List[str]:
+    def split_text(
+        self,
+        text: str,
+        allowed_special: Union[Literal["all"], AbstractSet[str]] = set(),
```

Comment:
thoughts on putting this in the init, eg `self.allowed_special` etc?

would be more consistent with other splitters (taking all the params at init rather than run time)

if you agree, im also happy to make this change myself

Reply:
hey Harrison, yes this makes sense, please feel free to make the change, thanks!